docs/internals/data-structures.rst

```diff
@@ -294,22 +294,180 @@ More on how this helps security in :ref:`security_structural_auth`.
 The manifest
 ~~~~~~~~~~~~
 
-The manifest is an object with an all-zero key that references all the
-archives. It contains:
+The manifest is the root of the object hierarchy. It references
+all archives in a repository, and thus all data in it.
+Since no object references it, it cannot be stored under its ID key.
+Instead, the manifest has a fixed all-zero key.
 
-* Manifest version
-* A list of archive infos
-* timestamp
-* config
+The manifest is rewritten each time an archive is created, deleted,
+or modified. It looks like this:
 
-Each archive info contains:
+.. code-block:: python
+
+    {
+        b'version': 1,
+        b'timestamp': b'2017-05-05T12:42:23.042864',
+        b'item_keys': [b'acl_access', b'acl_default', ...],
+        b'config': {},
+        b'archives': {
+            b'2017-05-05-system-backup': {
+                b'id': b'<32 byte binary object ID>',
+                b'time': b'2017-05-05T12:42:22.942864',
+            },
+        },
+        b'tam': ...,
+    }
+
+The *version* field can be either 1 or 2. The versions differ in the
+way feature flags are handled, described below.
+
+The *timestamp* field was used to avoid a certain class of replay attack.
+It is still used for that purpose, however, the newer replay protection
+introduced in Borg 1.1 includes all reply attacks. Thus it is not strictly
+necessary any more.
```

Comment:
nonces manager only protects against nonce reuse (it does not alert when the nonce is going down, and the nonce value can stay even if everything else is rolled back because it is not encrypted). It does not protect against rollback of the logical repostory state. I think the timestamp is still the only protections against a server that just silently discards all new transactions. (of sometimes does a rollback to remove certain archives)